web/src/user_profile.js

```diff
@@ -284,6 +355,45 @@ export function register_click_handlers() {
         e.preventDefault();
     });
 
+    $("body").on("click", "#user-profile-modal .add-subscription-button", (e) => {
+        e.preventDefault();
+        const stream_id = Number.parseInt(user_profile_subscribe_widget.value(), 10);
+        const sub = sub_store.get(stream_id);
+        const $alert_box = $("#user-profile-streams-tab .stream_list_info");
+        function addition_success(data) {
+            if (Object.keys(data.subscribed).length > 0) {
+                ui_report.success(
+                    $t_html({defaultMessage: "subscribed successfully!"}),
+                    $alert_box,
+                    1200,
+                );
+            } else {
+                ui_report.client_error(
+                    $t_html({defaultMessage: "Already subscribed."}),
+                    $alert_box,
+                    1200,
+                );
+            }
+        }
+        function addition_failure() {
+            let error_message;
+            if (people.is_my_user_id(user_id)) {
+                error_message = $t(
+                    {defaultMessage: "Error in subscribing to #{stream_name}"},
+                    {stream_name: sub.name},
+                );
+            } else {
+                error_message = $t(
+                    {defaultMessage: "Error adding user to #{stream_name}"},
+                    {stream_name: sub.name},
+                );
+            }
```

Comment:
Let's not add two new strings for these cases; do we even need an `error_message`, or could we just display the error the server sent?

Reply:
So if you use what the server sends then this is one of the cases I tested where in two window, I deactivated a user and then in the other I tried to subscribe that user to a stream, so for a general user the error shouldn't be like the below one (what the server has sent), instead we should give proper errors, that's why I added these, do let me know if you want me to make changes.


<img width="524" alt="Screenshot 2023-08-23 at 11 41 22 PM" src="https://github.com/zulip/zulip/assets/66828942/d4ccc226-cacb-4a5d-a349-eb8471401725">

Reply:
I agree that outcome is bad. I think the bug there is that `principal_to_user_profile` needs new error handling. Opened https://github.com/zulip/zulip/issues/26593 for the API bug there, but I think with those changes, we should just show the server error messages here; I think it's pretty unlikely for this code to fail in practice and it seems like a waste to ask translators to translate two additional strings for it.